### PR TITLE
feat(ui): add clone action to machine slice

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Added clone action to machine slice

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open the clone form from the machine list and submit the form
- Check that Redux actions are dispatched and a response from the API is recorded (either `cloneSuccess` or `cloneError`). Ignore the unformatted errors for now as they'll be handled in follow-ups.
- Do the same from a machine details page.

## Fixes

Fixes canonical-web-and-design/app-squad#198

### CODE REVIEW

File: ui/src/app/store/machine/reducers.test.ts [hunk 1]
```diff
@@ -522,24 +523,92 @@ describe("machine reducer", () => {
     );
   });
 
-  it("reduces createError", () => {
```
Comment (Caleb-Ellis) [inline ui/src/app/store/machine/reducers.test.ts:525]: This was a duplicate in the test file

File: ui/src/app/machines/components/ActionFormWrapper/CloneForm/CloneForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        </MemoryRouter>
+      </Provider>
+    );
+
+    wrapper.find("Formik").invoke("onSubmit")({
+      interfaces: true,
+      source: "ghi789",
+      storage: false,
+    });
+    await waitForComponentToPaint(wrapper);
```
Comment (huwshimi) [inline ui/src/app/machines/components/ActionFormWrapper/CloneForm/CloneForm.test.tsx:56]: Is this required when using `invoke` above? I thought that handled the async bits, but maybe it still has to wait.

**Review by huwshimi — APPROVED** [review]
Looks good, just one little question.

File: ui/src/app/machines/components/ActionFormWrapper/CloneForm/CloneForm.test.tsx (hunk 2 above)
Comment (Caleb-Ellis) [reply ui/src/app/machines/components/ActionFormWrapper/CloneForm/CloneForm.test.tsx]: Oh yeah you're right, not sure why I included it!

**Pull request merged**